lib/JSON/Validator/Error.pm

```diff
@@ -34,11 +110,44 @@ L<JSON::Validator>.
 
 =head1 ATTRIBUTES
 
+=head2 details
```

Comment:
I find this style of documentation to be really confusing. It's implying that the user should be assigning to details -- when actually, the user will only calling `details` as a getter, and will be returned an arrayref as a result. Indeed even the internals never use `details` as a setter -- this attribute is populated directly by the constructor.

Reply:
But it is perfectly fine to assign to it, if you want to make your own custom error, but in most cases you only want to read from it. I do welcome a better description that mention that though, but I fail to make up a short enough version.